File: app/src/main/java/com/kakacat/minitool/epidemicinquiry/adapter/DomesticAdapter.java
package com.kakacat.minitool.epidemicinquiry.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseExpandableListAdapter;
import android.widget.TextView;

import com.google.android.material.button.MaterialButton;
import com.kakacat.minitool.R;
import com.kakacat.minitool.common.util.SystemUtil;
import com.kakacat.minitool.epidemicinquiry.model.ChildBean;
import com.kakacat.minitool.epidemicinquiry.model.GroupBean;

import java.util.List;

public class DomesticAdapter extends BaseExpandableListAdapter {

    private List<GroupBean> groupBeanList;
    private LayoutInflater inflater;

    public DomesticAdapter(Context context, List<GroupBean> groupBeanList) {
        this.inflater = LayoutInflater.from(context);
        this.groupBeanList = groupBeanList;
    }

    @Override
    public int getGroupCount() {
        SystemUtil.log("getGroupCount" + groupBeanList.size());
        return groupBeanList.size();
    }

    @Override
    public int getChildrenCount(int i) {
        SystemUtil.log("getChildrenCount" + groupBeanList.get(i).getChildBeanList().size());
        return groupBeanList.get(i).getChildBeanList().size();
    }

    @Override
    public Object getGroup(int i) {
        SystemUtil.log("getGroup");
        return groupBeanList.get(i);
    }

    @Override
    public Object getChild(int i, int i1) {
        SystemUtil.log("getChild");
        return groupBeanList.get(i).getChildBeanList().get(i1);
    }

    @Override
    public long getGroupId(int i) {
        return i;
    }

    @Override
    public long getChildId(int i, int i1) {
        return i;
    }

    @Override
    public boolean hasStableIds() {
        return true;
    }

    @Override
    public View getGroupView(int i, boolean b, View view, ViewGroup viewGroup) {
        GroupViewHolder holder;
        GroupBean groupBean = groupBeanList.get(i);
        if (view == null) {
            view = inflater.inflate(R.layout.group_layout, viewGroup, false);
            holder = new GroupViewHolder(view);
            view.setTag(holder);
        } else {
            holder = (GroupViewHolder) view.getTag();
        }

        holder.btnLocation.setText(groupBean.getLocation());
        holder.tvCurrentConfirmCount.setText(groupBean.getCurrentConfirmCount());
        holder.tvSuspectCount.setText(groupBean.getSusPectCount());
        holder.tvCuredCount.setText(groupBean.getCuredCount());
        holder.tvTotalConfirmCount.setText(groupBean.getTotalConfirmCount());
        holder.tvDeadCount.setText(groupBean.getDeadCount());

        return view;
    }

    @Override
    public View getChildView(int i, int i1, boolean b, View view, ViewGroup viewGroup) {
        ChildViewHolder holder;
        ChildBean childBean = groupBeanList.get(i).getChildBeanList().get(i1);

        if (view == null) {
            view = inflater.inflate(R.layout.child_layout, viewGroup, false);
            holder = new ChildViewHolder(view);
            view.setTag(holder);
        } else {
            holder = (ChildViewHolder) view.getTag();
        }

        holder.tvLocation.setText(childBean.getCityName());
        holder.tvCurrentConfirmCount.setText(childBean.getCurrentConfirmedCount());
        holder.tvCuredCount.setText(childBean.getCuredCount());
        holder.tvSuspectCount.setText(childBean.getSusPectedCount());
        holder.tvDeadCount.setText(childBean.getDeadCount());
        holder.tvTotalConfirmCount.setText(childBean.getConfirmedCount());

        return view;
    }

    @Override
    public boolean isChildSelectable(int i, int i1) {
        return true;
    }

    static class GroupViewHolder {

        private MaterialButton btnLocation;
        private TextView tvCurrentConfirmCount;
        private TextView tvTotalConfirmCount;
        private TextView tvSuspectCount;
        private TextView tvCuredCount;
        private TextView tvDeadCount;

        public GroupViewHolder(View itemView) {
            btnLocation = itemView.findViewById(R.id.btn_location);
            tvCurrentConfirmCount = itemView.findViewById(R.id.tv_current_confirm_count);
            tvTotalConfirmCount = itemView.findViewById(R.id.tv_total_confirm_count);
            tvSuspectCount = itemView.findViewById(R.id.tv_suspect_count);
            tvCuredCount = itemView.findViewById(R.id.tv_cured_count);
            tvDeadCount = itemView.findViewById(R.id.tv_dead_count);
        }
    }

    static class ChildViewHolder {

        private TextView tvLocation;
        private TextView tvCurrentConfirmCount;
        private TextView tvTotalConfirmCount;
        private TextView tvSuspectCount;
        private TextView tvCuredCount;
        private TextView tvDeadCount;

        public ChildViewHolder(View itemView) {
            tvLocation = itemView.findViewById(R.id.tv_location);
            tvCurrentConfirmCount = itemView.findViewById(R.id.tv_current_confirm_count);
            tvTotalConfirmCount = itemView.findViewById(R.id.tv_total_confirm_count);
            tvSuspectCount = itemView.findViewById(R.id.tv_suspect_count);
            tvCuredCount = itemView.findViewById(R.id.tv_cured_count);
            tvDeadCount = itemView.findViewById(R.id.tv_dead_count);
        }
    }
}

File: app/src/main/java/com/kakacat/minitool/epidemicinquiry/adapter/DomesticAdapter.kt
package com.kakacat.minitool.epidemicinquiry.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.BaseExpandableListAdapter
import android.widget.TextView
import com.google.android.material.button.MaterialButton
import com.kakacat.minitool.R
import com.kakacat.minitool.common.util.SystemUtil.log
import com.kakacat.minitool.epidemicinquiry.model.GroupBean

class DomesticAdapter(private val groupBeanList: List<GroupBean>) : BaseExpandableListAdapter() {

    private var inflater: LayoutInflater? = null

    override fun getGroupCount(): Int {
        log("getGroupCount" + groupBeanList.size)
        return groupBeanList.size
    }

    override fun getChildrenCount(i: Int): Int {
        log("getChildrenCount" + groupBeanList[i].childBeanList!!.size)
        return groupBeanList[i].childBeanList!!.size
    }

    override fun getGroup(i: Int): Any {
        log("getGroup")
        return groupBeanList[i]
    }

    override fun getChild(i: Int, i1: Int): Any {
        log("getChild")
        return groupBeanList[i].childBeanList!![i1]
    }

    override fun getGroupId(i: Int): Long {
        return i.toLong()
    }

    override fun getChildId(i: Int, i1: Int): Long {
        return i.toLong()
    }

    override fun hasStableIds(): Boolean {
        return true
    }

    override fun getGroupView(i: Int, b: Boolean, view: View?, viewGroup: ViewGroup): View {
        if(inflater == null){
            inflater = LayoutInflater.from(viewGroup.context)
        }

        var contentView = view
        val holder: GroupViewHolder
        val groupBean = groupBeanList[i]
        if (contentView == null) {
            contentView = inflater!!.inflate(R.layout.group_layout, viewGroup, false)
            holder = GroupViewHolder(contentView)
            contentView.tag = holder
        } else {
            holder = contentView.tag as GroupViewHolder
        }
        holder.btnLocation.text = groupBean.location
        holder.tvCurrentConfirmCount.text = groupBean.currentConfirmCount
        holder.tvSuspectCount.text = groupBean.susPectCount
        holder.tvCuredCount.text = groupBean.curedCount
        holder.tvTotalConfirmCount.text = groupBean.totalConfirmCount
        holder.tvDeadCount.text = groupBean.deadCount
        return contentView!!
    }

    override fun getChildView(i: Int, i1: Int, b: Boolean, view: View?, viewGroup: ViewGroup): View {
        var contentView = view
        val holder: ChildViewHolder
        val childBean = groupBeanList[i].childBeanList!![i1]

        if(contentView == null){
            contentView = inflater!!.inflate(R.layout.child_layout,viewGroup,false)
            holder = ChildViewHolder(contentView)
            contentView.tag = holder
        }else{
            holder = contentView.tag as ChildViewHolder
        }

        holder.tvLocation.text = childBean.cityName
        holder.tvCurrentConfirmCount.text = childBean.currentConfirmedCount
        holder.tvCuredCount.text = childBean.curedCount
        holder.tvSuspectCount.text = childBean.susPectedCount
        holder.tvDeadCount.text = childBean.deadCount
        holder.tvTotalConfirmCount.text = childBean.confirmedCount

        return contentView!!
    }

    override fun isChildSelectable(i: Int, i1: Int): Boolean {
        return true
    }

    internal class GroupViewHolder(itemView: View) {
        val btnLocation: MaterialButton = itemView.findViewById(R.id.btn_location)
        val tvCurrentConfirmCount: TextView = itemView.findViewById(R.id.tv_current_confirm_count)
        val tvTotalConfirmCount: TextView = itemView.findViewById(R.id.tv_total_confirm_count)
        val tvSuspectCount: TextView = itemView.findViewById(R.id.tv_suspect_count)
        val tvCuredCount: TextView = itemView.findViewById(R.id.tv_cured_count)
        val tvDeadCount: TextView = itemView.findViewById(R.id.tv_dead_count)

    }

    internal class ChildViewHolder(itemView: View) {
        val tvLocation: TextView = itemView.findViewById(R.id.tv_location)
        val tvCurrentConfirmCount: TextView = itemView.findViewById(R.id.tv_current_confirm_count)
        val tvTotalConfirmCount: TextView = itemView.findViewById(R.id.tv_total_confirm_count)
        val tvSuspectCount: TextView = itemView.findViewById(R.id.tv_suspect_count)
        val tvCuredCount: TextView = itemView.findViewById(R.id.tv_cured_count)
        val tvDeadCount: TextView = itemView.findViewById(R.id.tv_dead_count)
    }

}